internal/podman/podman.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podman
+
+import (
+	"encoding/json"
+	"io"
+	"os/exec"
+	"strconv"
+)
+
+type Opts struct {
+	Debug     bool
+	OutWriter io.Writer
+}
+
+type RunContainerOpts struct {
+	Detach   bool
+	Image    string
+	Name     string
+	Hostname string
+	// map[hostVolume, pathInContainer]
+	Volumes map[string]string
+	// map[hostPort, containerPort]
+	Ports   map[int]int
+	Network string
+	EnvVars map[string]string
+	Args    []string
+}
+
+type Container struct {
+	Names []string `json:"Names"`
+	State string   `json:"State"`
+	Image string   `json:"Image"`
+	Ports []struct {
+		HostPort      int `json:"host_port"`
+		ContainerPort int `json:"container_port"`
+	} `json:"Ports,omitempty"`
+}
+
+func (o *Opts) runPodman(arg ...string) ([]byte, error) {
+	cmd := exec.Command("podman", arg...)
```

Comment:
I though they had some sort of programmatic API :(

Reply:
they have some bindings https://github.com/containers/podman/blob/main/pkg/bindings/README.md
but not sure if this will work in windows.
I created a separate investigation task for this https://jira.mongodb.org/browse/CLOUDP-196392